### ISSUE TITLE
Question: GCN pytorch implementation

### ISSUE THREAD

**Issue by byteSamurai** [dsgiitr/graph_nets#13]
Thanks a lot for the code. Found it via your blog post about GCNs.

I see Kipf used Laplace symmetric normalisation - as you wrote also here: https://dsgiitr.com/blogs/gcn/ 

But I don't understand, why the diagonal degree matrix is produced using A instead of A_hat:
```
        self.A_hat = A+torch.eye(A.size(0))
        self.D     = torch.diag(torch.sum(A,1))  #<< should this not be self.A_hat?
        self.D     = self.D.inverse().sqrt()
```

**Comment by AnirudhDagar** [comment]
Hi @byteSamurai,

Thanks for pointing this out, really appreciate it. Technically the following is correct:
```python
self.D = torch.diag(torch.sum(A_hat,1))
``` 
Would you like to send in a PR fixing the issue?

**Comment by byteSamurai** [comment]
sure. just a minute
